packages/core-toggle/core-toggle.test.js

```diff
@@ -1,167 +1,181 @@
-  t.true(await page.$eval('button#outer + core-toggle', el => el.hasAttribute('hidden')))
-})
+  it('opens and closes nested toggle', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <button id="outer">Toggle outer</button>
+        <core-toggle hidden>
+          <button id="inner">Toggle inner</button>
+          <core-toggle hidden>
+            <div>Inner content</div>
+          </core-toggle>
+        </core-toggle>
+      `
+    })
+    await $('button#outer').click()
+    await $('button#inner').click()
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await $('button#inner').click()
+    await expect($('button#inner + core-toggle').getAttribute('hidden')).toEqual('true')
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await $('button#outer').click()
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual('true')
+  })
 
-test('closes on outside click with popup', withPage, async (t, page) => {
-  await page.setContent(`
-    <button>Toggle</button>
-    <core-toggle popup hidden></core-toggle>
-  `)
-  await page.click('button')
-  t.false(await page.$eval('core-toggle', el => el.hasAttribute('hidden')))
-  await page.click('body')
-  t.true(await page.$eval('core-toggle', el => el.hasAttribute('hidden')))
-})
+  it('closes nested toggle with esc', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <button id="outer">Toggle outer</button>
+        <core-toggle hidden>
+          <button id="inner">Toggle inner</button>
+          <core-toggle hidden>
+            <div>Inner content</div>
+          </core-toggle>
+        </core-toggle>
+      `
+    })
+    await $('button#outer').click()
+    await $('button#inner').click()
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await $('button#inner').sendKeys(protractor.Key.ESCAPE)
+    await expect($('button#inner + core-toggle').getAttribute('hidden')).toEqual('true')
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual(null)
+    await $('button#inner').sendKeys(protractor.Key.ESCAPE)
+    await expect($('button#outer + core-toggle').getAttribute('hidden')).toEqual('true')
+  })
 
-test('respects "for" attribute', withPage, async (t, page) => {
-  await page.setContent(`
-    <div><button for="content">Toggle</button></div>
-    <core-toggle id="content" hidden></core-toggle>
-  `)
-  t.true(await page.$eval('core-toggle', el => el.button.getAttribute('for') === el.id))
-  t.true(await page.$eval('core-toggle', el => el.button.getAttribute('aria-controls') === el.id))
-})
+  it('closes on outside click with popup', async () => {
```

Comment:
Wording on this test is a bit confusing :p. Perhaps more along the lines of "closes popup when click outside"?

Reply:
yes, confusing. fixing